### PR TITLE
Git: Make the logic to create a missing local tag more robust

### PR DESCRIPTION
**Pull request by sschuberth**
As parsing human-readable output is generally fragile, instead check
whether the revision is a remote tag in order to conditionally create a
matching local tag.

Also, as right now (before the scan-by-repository topic is implemented)
there could be race conditions with multiple paths from a single Git
repository being downloaded in parallel, force to create the local tag
in case it already exists.

Signed-off-by: Sebastian Schuberth <sebastian.schuberth@bosch.io>

### CODE REVIEW

**Comment by sschuberth** [comment]
@heliocastro can you please check whether this makes your bogus `fatal: tag '1.1.1' already exists` output go away?

**Review by MarcelBochtler — APPROVED** [review]

**Comment by sschuberth** [comment]
The rationale was wrong, and I had to tweak the implementation slightly.

File: downloader/src/main/kotlin/vcs/Git.kt [hunk 1]
```diff
@@ -183,17 +182,13 @@ class Git : VersionControlSystem(), CommandLineTool {
...
-            // not fetched. So create it manually afterwards.
-            val tag = fetch.stderr.lineSequence().mapNotNull {
-                it.withoutPrefix(" * tag ")?.substringBefore("->")?.trim()
-            }.firstOrNull()
-
-            if (revision == tag) {
-                workingTree.runGit("tag", revision, "FETCH_HEAD")
+            // not fetched. So create the local tag manually afterwards.
+            if (revision in workingTree.listRemoteTags()) {
+                workingTree.runGit("tag", "-f", revision, "FETCH_HEAD")
```
Comment (MarcelBochtler) [inline downloader/src/main/kotlin/vcs/Git.kt:191]: Do you want to mention your rational why the `-f` flat is required in the comment above?
IMO it's important enough to not _only_ be in the commit message.

**Review by MarcelBochtler — APPROVED** [review]

**Review by heliocastro — APPROVED** [review]
Works for me

File: downloader/src/main/kotlin/vcs/Git.kt (hunk 1 above)
Comment (sschuberth) [reply downloader/src/main/kotlin/vcs/Git.kt:191]: > IMO it's important enough to not _only_ be in the commit message.

I think it's fine 😛 More seriously, it's temporary eventually anyway, as the `-f` could be removed once scan-by-repo is implemented.

**Pull request merged**